omq rpc log subscription

tewinget opened the pull request:
respects whether RPC is enabled, removes the log sink otherwise

bumps oxen-mq and oxen-logging

tewinget committed:
omq rpc log subscription

respects whether RPC is enabled, removes the log sink otherwise

bumps oxen-mq and oxen-logging

jagerman committed:
Add ring buffer sink to macos build

jagerman committed:
Apple: fix build failure when file already exists

The tools to create a dmg on Apple are flakey, of course, and fail in
cryptic ways if the file already exists, so purge it in the
contrib/mac.sh script.

jagerman committed:
macos: Fix hanging --start when Lokinet already running

If you stop/start the GUI but it doesn't exit on start, the second
--start (when lokinet is already running) waits for a state change that
doesn't come (because lokinet is already running).  This add a check for
already-running so that we exit right away in such a case.

jagerman committed:
Lower log level for renewals to debug

Otherwise you see a lot of unnecessary subscription renewals every 30s
in the logs.

majestrate commented on llarp/rpc/rpc_server.cpp:
```diff
@@ -12,10 +12,17 @@
 {
-  RpcServer::RpcServer(LMQ_ptr lmq, AbstractRouter* r) : m_LMQ(std::move(lmq)), m_Router(r)
+  RpcServer::RpcServer(LMQ_ptr lmq, AbstractRouter* r)
+      : m_LMQ(std::move(lmq)), m_Router(r), log_subs(*m_LMQ, llarp::logRingBuffer)
```
curly brace initialization

majestrate commented on llarp/util/logging.hpp:
```diff
@@ -25,6 +26,8 @@ namespace llarp
 // Deprecated loggers (in the top-level llarp namespace):
 namespace llarp
 {
+  inline std::shared_ptr<log::RingBufferSink> logRingBuffer = nullptr;
```
is there a good reason to have this be a "global"?

jagerman replied:
We talked about this in more depth off github; the answer is that the logging system is inherently (already) using static globals because of how logging needs to done easily from anywhere: needing to carry around an instance to get to the logger makes logging much more painful.  As a result, the sink will already be held in static global variables; this is just storing another pointer to it here because it's a bit awkward to extract it from the main logging system internals.

jagerman committed:
fix macos assemble_gui dependency

The `sign` target on macos was not working properly -- the signing
script would run before the build is finished.  This was caused by
cmake/macos.cmake having an `if(BUILD_GUI)`, but BUILD_GUI isn't defined
as an option until cmake/gui.cmake, which hadn't been included yet where
macos.cmake was included.

This extracts just the `option(BUIL_GUI)` from gui.cmake into a separate
gui-option.cmake file that we can load earlier to fix it.

While here I also noticed the GUI_EXE setting was defined as an option,
but isn't actually a boolean value, as an option, but isn't actually a
boolean value, so fixed it by making it a `set(... CACHE FILEPATH ...)`.

jagerman committed:
Bump lokinet-gui to latest

For log support, and other various changes/improvements.
